server/routes/journeys/manage/visit/handlers/timeSlotHandler.ts

```diff
@@ -2,46 +2,45 @@ import { Request, Response } from 'express'
 import { Page } from '../../../../../services/auditService'
 import { PageHandler } from '../../../../interfaces/pageHandler'
 import OfficialVisitsService from '../../../../../services/officialVisitsService'
+import { getParsedDateFromQueryString, getWeekOfDatesStartingMonday } from '../../../../../utils/utils'
+import { schema } from './timeSlotSchema'
 
 export default class TimeSlotHandler implements PageHandler {
   public PAGE_NAME = Page.CHOOSE_TIME_SLOT_PAGE
 
   constructor(private readonly officialVisitsService: OfficialVisitsService) {}
 
+  BODY = schema
+
   public GET = async (req: Request, res: Response) => {
-    const dayCode = (req.query['dayCode'] as string) || 'MON'
-    const timeSlots = await this.officialVisitsService.getAvailableSlots(
-      res,
-      'MDI',
-      new Date().toISOString().substring(0, 10),
-      new Date(Date.now() + 1000 * 60 * 60 * 24 * 7).toISOString().substring(0, 10),
-    )
-    req.session.journey.officialVisit.availableSlots = timeSlots
-
-    const schedule = await this.officialVisitsService.getSchedule(res, 'MDI', new Date().toISOString().substring(0, 10))
-
-    res.render('pages/manage/chooseTimeSlot', {
+    const { date = '' } = req.query
+    const selectedDate = getParsedDateFromQueryString(date.toString(), new Date())
+    const { weekOfDates, previousWeek, nextWeek } = getWeekOfDatesStartingMonday(selectedDate)
+
+    const timeSlots = await this.officialVisitsService.getAvailableSlots(res, 'MDI', selectedDate, selectedDate)
+    const { officialVisit } = req.session.journey
```

Comment:
Can the hard-coded `MDI` be replaced with the session value stored in the selected prisoner?